compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1173,8 +1168,10 @@ private boolean isXMLTypeAssignable(BType sourceType, BType targetType, Set<Type
                 }
                 return isAssignable(source.constraint, targetType, unresolvedTypes);
             }
-        } else if (sourceTag == TypeTags.XML_TEXT && targetTag == TypeTags.STRING) {
-            return true;
+        } else if (sourceTag == TypeTags.XML_TEXT) {
+            if (targetTag == TypeTags.STRING || targetTag == TypeTags.CHAR_STRING) {
```

Comment:
Is the second condition correct? This is only true if the XML text value is of length 1 which we can't guarantee here, right?

Reply:
This is just this https://github.com/ballerina-platform/ballerina-lang/pull/28548/files#diff-51851f1451506fedd7c81b67a92282fada4eb00fe4728c70445d7262aed401f3L825

Reply:
Yeah, but that's wrong, right? Shall we remove the check?

The following fails at the backend atm.

```ballerina
public function main() {
    xml:Text x = xml `foo`;
    string:Char y = x;
}
```

Created an issue too https://github.com/ballerina-platform/ballerina-lang/issues/28814

Reply:
Removed